src/Workspaces/CSharp/Portable/Classification/Worker.cs

```diff
@@ -80,10 +80,28 @@ private void ClassifyNodeOrToken(SyntaxNodeOrToken nodeOrToken)
 
     private void ClassifyNode(SyntaxNode node)
     {
-        foreach (var token in node.DescendantTokens(span: _textSpan, descendIntoTrivia: false))
+        using var _ = ArrayBuilder<SyntaxNode>.GetInstance(out var stack);
+        stack.Push(node);
+
+        while (stack.TryPop(out var current))
         {
             _cancellationToken.ThrowIfCancellationRequested();
-            ClassifyToken(token);
+
+            // It's ok that we're not pushing in reverse.  The caller (TotalClassificationTaggerProvider) will be
+            // sorting the results before doing anything with them.
+            foreach (var child in current.ChildNodesAndTokens())
+            {
+                if (child.IsNode)
```

Comment:
>if (child.IsNode) [](http://example.com/codeflow?start=16&length=17)

nit question: I've seen this in a couple places, why not just use the "bool AsNode(out SyntaxNode? node)" method?

Reply:
If possible, I would prefer it be called `IsNode(out SyntaxNode?)`. I don't think this is possible with a property and method in the same type, but may be possible with it defined as an extension method.

Reply:
I didn't know the AsNode method existed. Will update to that.

Reply:
done.